File: apps/desktop/src/state/shellIntegration.ts
import { invoke } from "@tauri-apps/api/core";

export type Shell = "zsh" | "bash" | "fish";
export type Status = "no_rcfile" | "not_installed" | "current" | "outdated";

export interface ShellStatus {
  shell: Shell;
  rcfile: string;
  status: Status;
}

export const fetchStatus = (): Promise<ShellStatus[]> =>
  invoke<ShellStatus[]>("shell_integration_status");

export const install = (shell: Shell): Promise<ShellStatus> =>
  invoke<ShellStatus>("shell_integration_install", { shell });

export const uninstall = (shell: Shell): Promise<ShellStatus> =>
  invoke<ShellStatus>("shell_integration_uninstall", { shell });

export const fetchSnippet = (shell: Shell): Promise<string> =>
  invoke<string>("shell_integration_snippet", { shell });

File: apps/desktop/src-tauri/src/shell_integration.rs
//! Tauri command surface for shell-integration install / uninstall / status.

use blaze_shell_integration::{self as si, Shell, ShellStatus};

#[tauri::command]
pub fn shell_integration_status() -> Vec<ShellStatus> {
    si::status_all()
}

#[tauri::command]
pub fn shell_integration_install(shell: Shell) -> Result<ShellStatus, String> {
    si::install(shell).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn shell_integration_uninstall(shell: Shell) -> Result<ShellStatus, String> {
    si::uninstall(shell).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn shell_integration_snippet(shell: Shell) -> &'static str {
    si::snippet(shell)
}
